MGMT-5379: Installation progress indication

DanielleChason opened the pull request:
https://issues.redhat.com/browse/MGMT-5379

I implemented Nir's new design for the installation progress bar. It contains more informative details for the user. You can see the main behaviors in the screen shots attached here. 

Note: some of the screen shot might look a little bit odd, it due to the limitation of the host+cluster mocking I used. 

Note 2: the button "send feedback" in the "instllation success alerts" does nothing at the moment, @nirfarkas will connect it to the feedback system. 

**Screenshots**: 

![image](https://user-images.githubusercontent.com/69599321/157471198-c24425aa-b590-425a-88e7-e2d2cfaa3e02.png)

![image](https://user-images.githubusercontent.com/69599321/157471289-7c17d08d-850b-4b46-aeca-8a2ce2b9e0ed.png)

![image](https://user-images.githubusercontent.com/69599321/157471344-15959ba5-26dc-4fbd-8ef7-74fd5957ec31.png)

![image](https://user-images.githubusercontent.com/69599321/157470544-b96974f6-deae-4ae8-b1e4-eef7d64108c7.png)

![image](https://user-images.githubusercontent.com/69599321/157470753-6ea93902-3a17-4cc5-9654-7b769573ac0b.png)

![image](https://user-images.githubusercontent.com/69599321/157470879-73f100c9-d915-4a8c-afe2-48986566087d.png)

![image](https://user-images.githubusercontent.com/69599321/157470934-4ead3eec-ac2f-441a-b260-01c37bd074cf.png)

![image](https://user-images.githubusercontent.com/69599321/157471044-1f858851-54e5-46b4-b6bc-1bd25ac68639.png)

![image](https://user-images.githubusercontent.com/69599321/157471121-96a0967c-b320-4578-88cd-53a7eba7f3f2.png)

![image](https://user-images.githubusercontent.com/69599321/157471424-d11d853f-a0e5-4488-951c-6edac2d6a3e6.png)

nirfarkas commented:
@DanielleChason This looks awesome!
I have some comments and tweaks we need to discuss before approving though (:

jtomasek commented on src/common/components/ui/BorderedIcon/BorderedIcon.tsx:
```diff
@@ -0,0 +1,20 @@
+  border: '1px solid',
+  borderColor: greyColor.value,
+  boxSizing: 'content-box',
+  verticalAlign: '-1.125em',
```
I checked how patternfly determines vertical align per icon size and for our icon size (including border) the vertical align should be `-0.3em;`

jtomasek replied:
To make the component universal for all icon sizes we should calculate the padding based on the `size` prop which we add to this component (sm, md, lg, xl) Similar to [1]. Same for the verticalAlign.

[1] https://github.com/patternfly/patternfly-react/blob/main/packages/react-icons/src/createIcon.tsx#L10

jtomasek commented on src/common/components/ui/BorderedIcon/BorderedIcon.tsx:
```diff
@@ -0,0 +1,20 @@
+import { DownloadIcon } from '@patternfly/react-icons';
+import { global_Color_200 as greyColor } from '@patternfly/react-tokens';
+
+const style = {
```
```suggestion
const style: React.CSSProperties = {
```

nirfarkas commented:
@jtomasek You're a magician!

DanielleChason commented:
I fixed the icons (@jtomasek helped me a lot! Thank you!) and now there is a better spacing between the progress bar items (though it is still limited and won't work in all of screen sizes. I think that it is currently the best way to it, taking into account the limitation of Patternfly). 
Also I fixed all the other issues.

**Screenshots:** 
![image](https://user-images.githubusercontent.com/69599321/158423501-48e283ec-3400-4262-b2cd-e44b9375f3ec.png)
![image](https://user-images.githubusercontent.com/69599321/158423562-436c041f-f697-489b-81d0-c190161f8698.png)
![image](https://user-images.githubusercontent.com/69599321/158423618-563ef91a-6207-4099-aa8f-3ee3608332e3.png)

nirfarkas commented:
Under the "Control plane" it should say "Control plane nodes..."
Other than that looks perfect!

jkilzi commented:
@DanielleChason PTAL there are some merge conflicts

jtomasek commented on src/common/components/clusterDetail/ProgressBarTexts.tsx:
```diff
@@ -0,0 +1,86 @@
+        )}`;
+  const getHostName = (hostRole: HostRole): React.ReactElement => {
+    if (hostRole == 'master') {
+      return <StackItem>Control Plane</StackItem>;
```
Redundant <StackItem>
```suggestion
      return 'Control Plane';
```

jtomasek commented on src/common/components/clusterDetail/ProgressBarTexts.tsx:
```diff
@@ -0,0 +1,86 @@
+    if (hostRole == 'master') {
+      return <StackItem>Control Plane</StackItem>;
+    }
+    return <StackItem> Workers</StackItem>;
```
same as above

jtomasek commented on src/common/components/clusterDetail/ProgressBarTexts.tsx:
```diff
@@ -0,0 +1,86 @@
+            </BorderedIcon>
+          </StackItem>
+          <StackItem>
+            {getHostName(hostRole)}
```
```suggestion
            {getHostName(hostRole)}<br/>
```

jtomasek commented on src/common/components/clusterDetail/ProgressBarTexts.tsx:
```diff
@@ -0,0 +1,86 @@
+            </BorderedIcon>
+          </StackItem>
+          <StackItem>
+            {getHostName(hostRole)}
```
```suggestion
            {getHostName(hostRole)}<br/>
```

jtomasek commented on src/common/components/clusterDetail/ProgressBarTexts.tsx:
```diff
@@ -0,0 +1,86 @@
+          </BorderedIcon>
+        </StackItem>
+        <StackItem>
+          {getHostName(hostRole)}
```
```suggestion
          {getHostName(hostRole)}<br/>
```

jtomasek commented on src/common/components/clusterDetail/FinalizingProgress.tsx:
```diff
@@ -0,0 +1,131 @@
+              'Initialization'
+            )}
+          </>
+        </StackItem>
+        <>{initializationStatus}</>
```
```suggestion
           <br/>
           {initializationStatus}
        </StackItem>
       
```

jtomasek commented on src/common/components/clusterDetail/OperatorsProgressItem.tsx:
```diff
@@ -101,21 +102,24 @@ type OperatorsProgressItemProps = {
         </OperatorsPopover>
-      </FlexItem>
-    </Flex>
+      </StackItem>
+      {label}
```
```suggestion
      <br/>
      {label}
      </StackItem>
      
```

jtomasek commented on src/ocm/components/clusterDetail/ClusterDetail.tsx:
```diff
@@ -72,13 +77,25 @@ const ClusterDetail: React.FC<ClusterDetailProps> = ({ cluster }) => {
               totalPercentage={cluster.progress?.totalPercentage || 0}
             />
           </GridItem>
+          <GridItem span={6}></GridItem>
```
Leftover: 
```suggestion
        
```

DanielleChason replied:
Without that the progress bar itself and the alerts below that are in the same line (each one of them takes half of the screen). Do you know how can I overcome that?

jtomasek commented on src/common/components/clusterDetail/ClusterProgress.tsx:
```diff
@@ -68,154 +47,20 @@ const getInstallationStatus = (
+export type ClusterProgressProps = {
   cluster: Cluster;
   minimizedView?: boolean;
   onFetchEvents: EventListFetchProps['onFetchEvents'];
```
onFetchEvents and fallbackEventsUrl are not used in ClusterProgress component any more.

jtomasek commented on src/common/components/clusterDetail/ClusterProgressItems.tsx:
```diff
@@ -0,0 +1,55 @@
+  minimizedView = false,
+  onFetchEvents,
+  fallbackEventsURL,
+}: ClusterProgressProps) => {
```
Please define a new `ClusterProgressItemsProps` type

jtomasek commented on src/common/components/clusterDetail/ClusterProgress.tsx:
```diff
@@ -68,154 +47,20 @@ const getInstallationStatus = (
-};
-
-type ClusterProgressProps = {
+export type ClusterProgressProps = {
```
No need to export, we need to create ClusterProgressItemsProps anyway (see below)

jtomasek commented:
I've created a PR https://github.com/DanielleChason/assisted-ui-lib/pull/2 against your branch which adds the new progress items into the CIM cluster progress component.
